src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMerge.java

```diff
@@ -95,21 +81,21 @@ void setHelper(Ghprb helper) {
     }
 
     @Override
-    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, final BuildListener listener) throws InterruptedException, IOException {
-        logger = listener.getLogger();
-        AbstractProject<?, ?> project = build.getProject();
-        if (build.getResult().isWorseThan(Result.SUCCESS)) {
-            logger.println("Build did not succeed, merge will not be run");
-            return true;
+    public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull Launcher launcher, @Nonnull TaskListener taskListener) throws InterruptedException, IOException {
```

Comment:
Sorry if I'm missing something obvious, but why is a filePath needed, and why can't it be `null`?